### DISCUSSION TITLE
Unknown option --goto

### DISCUSSION THREAD

**Discussion by alanhe421** [coder/code-server#5801] [Q&A]
![image](https://user-images.githubusercontent.com/9245110/203094989-05db7cbb-1ba9-4165-bbc3-7a4610f6d4c5.png)

Is there a mistake in the way it is used?

```sh
/usr/bin/code-server --auth none --goto /root/a.json@5:5
```
![image](https://user-images.githubusercontent.com/9245110/203095376-fd13d0ea-6785-4c5d-965e-bc176e395205.png)

## version info
code-server 4.8.3 977b853a1e162ab583aed64b1322d1515c57728c

**Comment by alanhe421** [comment]
@jsjoeio

Do you know the issue?

**Reply by jsjoeio** [reply]
Wait...is that a typo in the release notes? Does it work if you use `-r`? See here: https://github.com/coder/code-server/pull/5620 Hmm no that doesn't make sense. Let me check again.

**Reply by jsjoeio** [reply]
In my notes, that's what I did 🤔 Can you try that and report back? I may need to update release notes and changelog

**Reply by alanhe421** [reply]
![image](https://user-images.githubusercontent.com/9245110/205450106-b1ab6e98-3694-4f5a-b6ad-8008657a79a0.png)

**Reply by jsjoeio** [reply]
I can't tell but from the screenshot it looks like you're running that from within code-server or a VS Code Integrated Terminal. Is that right?

Here I do it outside and it works:

https://user-images.githubusercontent.com/3806031/205695607-764cf763-c288-4d50-a4e4-8923875df3c3.mov

Try this: `VSCODE_IPC_HOOK_CLI= code-server --auth none -r /root/a.json:2:3` 

OH actually wait...I just tried like you too. I think the `code-server -r` assumes you already have a running instance going so I don't think it will work like that 🤔 You'd have to do `code-server --auth none` and then later do `code-server -r <file>`. I don't know if that's considered a bug or not 🤔 cc @code-asher

**Reply by code-asher** [reply]
Oh oops yeah looks like we made a typo in the release notes.  No need for a new flag.  I think in VS Code it is a separate flag which is why we got confused.

Yeah, this is how it works by design.  The idea is that you spawn code-server as a system service that way it is always running in the background then `-r` will always try to connect to it and error if it is not up.

We could make `-r` spawn code-server if it has not spawned yet but it is kind of a weird scenario to be in because code-server is designed as a daemon and not a one-off process.

Also we plan on removing our code around this and calling Code's server CLI directly which has the same behavior so if we fix it now we would have to port those changes upstream somehow or maintain a patch. 

Still, if the behavior is confusing enough maybe we make the change anyway.

**Reply by code-asher** [reply]
Whatever we do, we should really improve that error.  ENOENT is not terribly helpful.

**Reply by alanhe421** [reply]
-r arg, from which version?


I test v4.7.1.  it always refreshes when -r file, and column/row is not work. 

it's different with v4.8.x.

**Reply by code-asher** [reply]
Yeah 4.8.0 added column/row support.  I believe the code for this is the same in 4.7.1 and 4.8.0 (other than column/row).  The logic goes like this:

1. Are we in an integrated terminal?  If yes try to open in the existing instance.
2. Are there any arguments set other than --reuse-window or --new-window set?  If yes spawn a new instance.
3. Does /tmp/vscode-ipc exist?  If yes try opening in the instance listed in that file.  If --reuse-window or --new-window are set do so unconditionally otherwise check first if the socket is actually in use and if not then spawn a new instance instead.
4. Otherwise spawn a new instance.

**Reply by code-asher** [reply]
I think we can simplify the logic and add some helpful output to make this easier to use.
